src/clr-angular/data/datagrid/render/render-organizer.spec.ts

```diff
@@ -31,13 +37,21 @@ export default function(): void {
             // Initial sizing to make sure clearWidths is included in the next one.
             this.organizer.resize();
             let step = 0;
-            this.organizer.noLayout.subscribe(on => expect(step++).toBe(on ? 0 : 7));
-            this.organizer.clearWidths.subscribe(() => expect(step++).toBe(1));
-            this.organizer.detectStrictWidths.subscribe(() => expect(step++).toBe(2));
-            this.organizer.tableMode.subscribe(on => expect(step++).toBe(on ? 3 : 5));
-            this.organizer.computeWidths.subscribe(() => expect(step++).toBe(4));
-            this.organizer.alignColumns.subscribe(() => expect(step++).toBe(6));
-            this.organizer.scrollbar.subscribe(() => expect(step++).toBe(8));
+            this.organizer.renderStep.subscribe(renderStep => {
+                if (renderStep === DatagridRenderStep.CALCULATE_MODE_ON) {
+                    expect(step++).toBe(0);
```

Comment:
I know this was an existing test, but I'm not a big fan of this `step++` business. It makes me have to stop and think, oh yeah its postfixed so its still `0` but after this expression will be `1`. 

Suggestion: move the `step + 1` addition to the top of the subscription since its done EVERY time.

Reply:
Will move.

Reply:
Actually, trying to quickly move this means I have to add logic that handles the 0 value case  for the first step and the increment case for the rest of the steps. 

I'll add a note to revisit the pattern if you are ok with it.